docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +164,104 @@ Note that:
   for instance `tedge/configuration_change//etc/tedge/mosquitto-conf/c8y-bridge.conf`.
 * Since the type of configuration file is used as an MQTT topic name, the characters `#` and `+` cannot be used in a type name.
   If such a character is used in a type name (or in the path of a configuration file without explicit type),
-  then the whole plugin configuration `/etc/tedge/c8y/c8y-configuration-plugin.toml` is considered ill-formed.
\ No newline at end of file
+  then the whole plugin configuration `/etc/tedge/c8y/c8y-configuration-plugin.toml` is considered ill-formed.
+
+## Configuration files for child devices
+
+To manage configuration files for child-devices the `c8y_configuration_plugin` supports two aspects:
+1) Allowing to associate a configuration file with a cloud's child-device twin
+2) Allowing to consume/provide a configuration file from/to an external device via network
+
+**Details to Aspect 1: Associating to cloud's child-device twin**
+
+For aspect (1) the plugin provides the field `childid` for all records in the `c8y_configuration_plugin` configuration (reference to section 'Configuration' above). That field is interpreted as unique child-device id and the plugin associates the record's configuration file with corresponding cloud's child-device twin. If not provided the configuration file is associated with cloud's thin-edge device twin.
+
+Example:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },              # appears in the cloud on the thin-edge device
+    { path = '/etc/child1/foo.conf', type = 'foo', childid = 'child1'  }  # appears in the cloud on the child-device 'child1'
+  ]
+```
+
+Note that the `c8y_configuration_plugin` does not create any child-device twin in the cloud. Instead the clouds child-device twin must be created upfront. However the plugin takes care to set all capabilities (supported operations, config-types) on that child-device twin in the cloud in the same way as it does for the thin-edge device.
+
+
+**Details to Aspect 2: Consuming/providing configuration files via network**
+
+For aspect (2) there are two proposals as below. Decision has to been taken which proposal to follow.
```

Comment:
That comment is collect pros and cons for all raised proposals.

List of the proposals:

**Proposal 1:** MQTT retain message (from the spec in that section, see [that line](https://github.com/thin-edge/thin-edge.io/pull/1236/files#diff-0d430102a9a7f4e4431390a85806a990b51d2143dbc97bf596a267c893e1f693R195)

**Proposal 2:** HTTP serviced by C8Y Config Plugin (from the spec in that section, see [that line](https://github.com/thin-edge/thin-edge.io/pull/1236/files#diff-0d430102a9a7f4e4431390a85806a990b51d2143dbc97bf596a267c893e1f693R228)

**Proposal 3:** HTTP serviced by tedge_agent (from Didiers comment, see https://github.com/thin-edge/thin-edge.io/pull/1236#discussion_r926708254)

Reply:
**Proposal 1:**
Pros: 
* simple to implement

Cons: 
* can consume much of RAM in case of large files or even many childs with many small files (note those retain messages are hold all along)


**Proposal 2:**
Cons:
* less re-use since implemented into/explecitely for the C8Y Config Plugin 
* far to complex to implement just for the C8Y Config Plugin


**Proposal 3:**
Pro:
* high re-use for (mostly) all thin-edge features (as logging, sw man)

Cons:
* Embedded Device has to deal with HTTP

Reply:
> I feel the plugin should handle file upload/download by itself. Maybe it needs the support of the MQTT bridge just to get the JWT token right?.

I feel your comment faces the file upload/download from the thin-edge device towards C8Y (the noth side). Instead those proposals from above are facing the file upload/download from the thin-edge device towards an external device (the south side); that could be a PLC or any other kind of Embedded Device.

So it is nothing about C8Y and it's JWT token, but about a file-transfer mechanism from thin-edge to any external device. Therefore the proposal 3 (HTTP serviced by tedge_agent) is now in focus.